docs/proposals/deployment.md

```diff
@@ -0,0 +1,119 @@
+  // if specified.
+  Template *PodTemplateSpec
+
+  // The deployment strategy to use to replace existing pods with new ones.
+  Strategy DeploymentStrategy
+}
+
+type DeploymentStrategy struct {
+  // Type of deployment. Can be "Recreate" or "RollingUpdate".
+  Type DeploymentType
+
+  // Rolling update config params. Present only if DeploymentType =
+  // RollingUpdate.
+  RollingUpdate *RollingUpdateDeploymentSpec
+}
+
+type DeploymentType string
+
+const (
+  // Kill all existing pods before creating new ones.
+  DeploymentRecreate DeploymentType = "Recreate"
+
+  // Replace the old RCs by new one using rolling update i.e gradually scale down the old RCs and scale up the new one.
+  DeploymentRollingUpdate DeploymentType = "RollingUpdate"
+)
+
+// Spec to control the desired behavior of rolling update.
+type RollingUpdateDeploymentSpec struct {
+  // The percentage of pod replicas that should be updated in each interval.
+  // If nil, a fixed value of 1 pod is used.
+  UpdatePercent *int
+
+  // While updating pods, a "readiness" check is done instead of just the
+  // pod "creation" check, if this is true. False, by default.
+  UseReadinessAsAcceptance bool
+}
+
+type DeploymentStatus struct {
+  // The total number of pods created with the desired template spec.
+  Replicas int
+
+  // The most recent generation observed by the controller.
+  ObservedGeneration int64
+}
+
+```
+
+### Controller
+
+The DeploymentController will make a given Deployment happen.
+Whenever a new Deployment is created, it will:
+
+1. Find all RCs with the same label selector as DeploymentSpec.Selector.
+2. Add annotations for bookkeeping:
+   - Add "deployment.kubernetes.io/<deployment-id>" annotation to the existing
+     RCs to identify the deployment controller that is controlling them.
+3. Create a new RC to create pods using DeploymentSpec.PodTemplate. Set replicas = 0.
+4. Add annotations to this new RC:
+   - Add "deployment.kubernetes.io/<deployment-id>" annotation to identify the deployment controller that is controlling that RC.
+   - Add "deployment.kubernetes.io/<generation>" annotation to identify the
+     deployment generation from which this RC was created. Deployment controller will
+     treat the RC with the same generation as DeploymentStatus.Generation as the
+     latest RC, which it needs to scale up.
+5. As per the deployment strategy, scale up the new RC and scale down the old ones.
+6. Cleanup.
+
+## Future
+
+Apart from the above, we want to add support for the following:
+* More DeploymentTypes: https://github.com/openshift/origin/blob/master/examples/deployment/README.md#deployment-types lists most commonly used ones.
+* Triggers: Deployment will have a trigger field to identify what triggered the deployment. Options are: Manual/UserTriggered, Autoscaler. Openshift deployments support triggers.
+* isInert: Allow Deployments to be marked as inert. Openshift has an active field, which indicates if deployments can be triggered or not.
```

Comment:
Inert may not be the right field - since the discussions about inert have been one way creation type things.  Perhaps this should be renamed since it's something that could happen at any point?